modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeCatchAllCheck.java

```diff
@@ -355,7 +355,9 @@ private String _formatGeneral(
 			String from = jsonObject.getString("from");
 			String to = jsonObject.getString("to");
 
-			if (from.contains(StringPool.OPEN_PARENTHESIS)) {
+			if (from.contains(StringPool.OPEN_PARENTHESIS) &&
+				to.contains(StringPool.OPEN_PARENTHESIS)) {
+
 				newContent = _formatMethodCall(
```

Comment:
hi @kyle-miho this will open the possibility of wrong replacements no?

The validation of the `formatMethodCall` returns if it doesn't have the right parameters and adding this clause will not validate the parameters anymore the to isn't a methodCall as well.

Only an example:

In the scenario that a class has an overload in a method, and one of these methods must be removed.

Something like:

```
from: Class.method(String,int)
to: ""
```

If the client code has `Class.method(int,String)` overload call, not the correct one. It's going to be removed incorrectly, isn't it?

Reply:
Please correct me if I'm wrong, but looking at it briefly, I had that interpretation. Be careful when changing the logic of UpgradeCatchAll, remember that it deals with hundreds of cases and we have to think of more possibilities impacted when we change its logic.

Reply:
> hi @kyle-miho this will open the possibility of wrong replacements no?
> 
> The validation of the `formatMethodCall` using `_hasValidMethodCall` returns if it doesn't have the right parameters and adding this clause will not validate the parameters anymore the to isn't a methodCall as well.
> 
> Only an example:
> 
> In the scenario that a class has an overload in a method, and one of these methods must be removed.
> 
> Something like:
> 
> ```
> from: Class.method(String,int)
> to: ""
> ```
> 
> If the client code has `Class.method(int,String)` overload call, not the correct one. It's going to be removed incorrectly, isn't it?

Please take a look at this before proceeding. 

Cc @drewbrokke

Reply:
@nicolas-sss I see this makes sense, unfortunately a lot of the newer cases are not implementable without changes to UpgradeCatchAll. I think instead we need to add an option to avoid parameter formatting with a flag? Does that sound okay instead? Since this case uses a hardcoded string instead of a variable

Reply:
Hi @kyle-miho it's a convention for the replacements file that if we do not check the parameters, we don't add them to the `from`. 
Once we add, we imagine that they should be checked, some exceptions are the case of the flag to avoid checking. Using will only check the quantity at the most.
But, don't get me wrong, I'm not again doing changes to the CatchAll, but the changes should follow the previous logic, this change will change the logic as shown in the example. 
And the example using a hardcore string it was a choice, could like:

```
"classNames": [
	"Class"
],
from: var.method(String,int)
to: ""
```

And the same problem will happen. 
Like, those cases worked before, adding changes to the CatchAll should only add new patterns to cover and not remove cases that worked once before.

Reply:
Unless it's going to bring enormous value, maybe, I don't know....
 But that's just one case, maybe there are other cases that we haven't seen yet that will also be impacted by removing the validation. What do you think about it?
Let's think a little bit more about this case. 
Do you have other ideas for this?